### PR TITLE
Update for Fastify v5

### PR DESCRIPTION
**Pull request by synapse**
Ref: https://github.com/fastify/fastify/issues/5116

- this PR replaces the embedded custom plugin in favor of 3 functions (maintaining the same structure inside) (tap 18 requires its plugins to be loaded from `node_modules`)
- considering the above change, I've replaced the plugin with subtests (that do not need a repetitive title) which will run the assertion function just as before

⚠️ ~~Not sure about the version of CI used for this repo. Currently is set to:~~
```
fastify/workflows/.github/workflows/plugins-ci-mongo.yml@7993e1ea858b5c3ceff1aa154fa44b892f311ac3
```
☝️ used version `v4.1.0`

~~Can anyone confirm that there's a newer version for deploying for this repo?~~

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

CC @simoneb

### CODE REVIEW

**Comment by simoneb** [comment]
> Not sure about the version of CI used for this repo.

Based on the diff, I'd say you can go to the same tag as the other repos

**Comment by simoneb** [comment]
it may be worth considering to do also #226 since it's a major. next branch created and rebased meanwhile

**Comment by climba03003** [comment]
FYI, the workflows hash uses because `mongodb@6` is dropped `node@14` supports earlier than `fastify@5` exists.
So, this plugin need to be dropping the support earlier than other plugin.

File: test/index.test.js [hunk 1]
```diff
@@ -16,55 +16,51 @@ const CLIENT_NAME = 'client_name'
 const ANOTHER_DATABASE_NAME = 'my_awesome_database'
 const COLLECTION_NAME = 'mycoll'
 
-t.Test.prototype.addAssert('objectId', 1, function (ObjectId, message, extra) {
+function objectIdTest (t, ObjectId, message) {
```
Comment (climba03003) [inline test/index.test.js:19]: If you plan to nest all this helper function inside `t.test`, please also specify the `t.plan` inside.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: test/index.test.js (hunk 1 above)
Comment (synapse) [reply test/index.test.js:19]: added `t.plan` in all 3 functions

**Review by climba03003 — APPROVED** [review]

**Comment by synapse** [comment]
> FYI, the workflows hash uses because `mongodb@6` is dropped `node@14` supports earlier than `fastify@5` exists. So, this plugin need to be dropping the support earlier than other plugin.

Hey @climba03003, I'm not sure if you've meant to keep the original hash or is it ok to move forward with the current `v4.1.0` tag?

**Comment by climba03003** [comment]
It just the background of why the hash exist, I am fine with `v4.1.0`

File: .taprc [hunk 2]
```diff
@@ -1,2 +1,3 @@
+disable-coverage: true
```
[the remark this answers is not in the document]
Comment (synapse) [reply .taprc]: You were right @gurgunday coverage is at 100%. I've removed disabling option from `.taprc`

**Review by gurgunday — APPROVED** [review]
lgtm

**Pull request merged**